### PR TITLE
Move maxengine for restructuring

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

[Restructure](https://github.com/AI-Hypercomputer/maxtext/blob/main/RESTRUCTURE.md) inference-related modules to align with the new `maxtext` directory.  Moved four files and created two `__init__.py` as below:
- `src/maxtext/inference/maxengine/`:
  - `__init___.py`.
  - `maxengine.py`
  - `maxengine_config.py`
  - `maxengine_server.py`
- `src/maxtext/inference/mlperf/microbenchmarks/`
  - `__init___.py`
  - `benchmark_chunked_prefill.py`
- Move `maxtext.decode` to `maxtext.inference.decode`.
- Add shim for both decode and maxengine_server under `MaxText`, with warning message for deprecation.
- Move folder `maxtext/inference/maxengine_server/` to `maxtext/inference/maxengine/maxengine_server_deployment/`. This [rename](https://screenshot.googleplex.com/9bbgbzHMprQ98qD) is to avoid naming conflict between this folder and the actual code in `maxengine_server.py`.
- Update imports/shell scripts/comments/docs throughout the repo.

# Tests

maxengine_server, ([logs](https://paste.googleplex.com/4800999555465216)):
```
# New API
python -m maxtext.inference.maxengine.maxengine_server maxtext/configs/base.yml model_name=qwen3-0.6b tokenizer_path=Qwen/Qwen3-0.6B load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-0.6b/scanned/2026-02-16-11-35/0/items scan_layers=true max_prefill_predict_length=1024 max_target_length=2048 ici_fsdp_parallelism=1 ici_autoregressive_parallelism=1 ici_tensor_parallelism=-1 weight_dtype=bfloat16 attention=dot_product per_device_batch_size=10 quantize_kvcache=False tokenizer_type=huggingface hf_access_token=xxx

# Shim API
python -m MaxText.maxengine_server maxtext/configs/base.yml model_name=qwen3-0.6b tokenizer_path=Qwen/Qwen3-0.6B load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-0.6b/scanned/2026-02-16-11-35/0/items scan_layers=true max_prefill_predict_length=1024 max_target_length=2048 ici_fsdp_parallelism=1 ici_autoregressive_parallelism=1 ici_tensor_parallelism=-1 weight_dtype=bfloat16 attention=dot_product per_device_batch_size=10 quantize_kvcache=False tokenizer_type=huggingface hf_access_token=xxx
```
decode:
```
# New API
python -m maxtext.inference.decode maxtext/configs/base.yml model_name=gemma3-4b tokenizer_path=google/gemma-3-4b-it tokenizer_type=huggingface load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/gemma3-4b/unscanned/001/0/items per_device_batch_size=1 run_name=ht_test max_prefill_predict_length=280 max_target_length=320 steps=1 async_checkpointing=false scan_layers=false use_multimodal=true prompt=\'Describe\ image\ \<start_of_image\>\' image_path=\'tests/assets/test_image.jpg\' attention=\'dot_product\' hf_access_token=xxx

# Shim API
python -m MaxText.decode maxtext/configs/base.yml model_name=gemma3-4b tokenizer_path=google/gemma-3-4b-it tokenizer_type=huggingface load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/gemma3-4b/unscanned/001/0/items per_device_batch_size=1 run_name=ht_test max_prefill_predict_length=280 max_target_length=320 steps=1 async_checkpointing=false scan_layers=false use_multimodal=true prompt=\'Describe\ image\ \<start_of_image\>\' image_path=\'tests/assets/test_image.jpg\' attention=\'dot_product\' hf_access_token=xxx
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/maxengine_test.py [hunk 1]
```diff
@@ -21,10 +21,11 @@
 import jax
 import jax.numpy as jnp
 from jax.sharding import Mesh
-from MaxText import maxengine, pyconfig
+from MaxText import pyconfig
 from MaxText.common_types import DECODING_ACTIVE_SEQUENCE_INDICATOR, MODEL_MODE_PREFILL
 from maxtext.layers import quantizations
-from MaxText.maxengine import MaxEngine
+from maxtext.inference.maxengine import maxengine
```
Comment (dipannita08) [inline tests/unit/maxengine_test.py:27]: Was this extra import intended?

File: src/maxtext/inference/decode.py [hunk 2]
```diff
@@ -22,10 +22,10 @@
...
 from absl import app
 
-from MaxText import maxengine
 from MaxText import pyconfig
 from maxtext.common import profiler
 from maxtext.common.gcloud_stub import jetstream, is_decoupled
+from maxtext.inference.maxengine import maxengine
 from maxtext.multimodal import processor as mm_processor
 from maxtext.multimodal import utils as mm_utils
 from maxtext.utils import max_utils
```
Comment (bvandermoon) [inline src/maxtext/inference/decode.py:31]: This isn't related to your PR, but can this be moved to `inference/`?

File: tests/unit/maxengine_test.py (hunk 1 above)
Comment (hengtaoguo) [reply tests/unit/maxengine_test.py:27]: Seems this duplicate `maxengine/MaxEngine` is a legacy problem. I've consolidated to one. Thanks for catching!

File: src/maxtext/inference/decode.py (hunk 2 above)
Comment (hengtaoguo) [reply src/maxtext/inference/decode.py:31]: Done, and added a shim at `MaxText.decode`

**Comment by hengtaoguo** [comment]
> Generally LGTM but I think there are a couple of `inference/` directories missing from [RESTRUCTURE.md](https://github.com/AI-Hypercomputer/maxtext/blob/3737eb35266833cbfffd96a143f126c0c41a6baa/RESTRUCTURE.md) like `offline/`, `user_config/` and `inference_microbenchmark.py/inference_microbenchmark_sweep.py` should go in `inference/microbenchmarks`. Could you please take a look?

SG I will write a follow-up PR moving other standalone files/folders under `maxtext.inference.mlperf`, otherwise the current PR will be too complicated to resolve dependencies. Thanks!
